### PR TITLE
Remove OpenSSL dependency from Windows and macOS et.al.

### PR DESCRIPTION
**Pull request by teo-tsirpanis**
See https://github.com/awslabs/aws-c-cal/blob/e4c50672c2cc4a6cd8904b79c8c0d7255d2d67f4/cmake/aws-c-cal-config.cmake#L5

Checklist
* [X] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [X] Bumped the build number (if the version is unchanged)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)

### CODE REVIEW

**Comment by teo-tsirpanis** [comment]
@conda-forge-admin, please rerender

**Comment by h-vetinari** [comment]
From looking at the link you provided, it seems we shouldn't remove the openssl dependency, but set `-DBYO_CRYPTO=ON` on all platforms? See [here](https://github.com/awslabs/aws-c-cal/blob/main/CMakeLists.txt).

**Comment by teo-tsirpanis** [comment]
My understanding is that `BYO_CRYPTO` is something completely different. As seen in `CMakeLists.txt` it prevents the platform-specific source files from being compiled, and [would cause the library to abort by default](https://github.com/awslabs/aws-c-cal/blob/e4c50672c2cc4a6cd8904b79c8c0d7255d2d67f4/source/hmac.c#L12-L20). Seems the use of `BYO_CRYPTO` is for embedded systems (https://github.com/awslabs/aws-c-cal/pull/1#discussion_r254418320 and https://github.com/aws/aws-iot-device-sdk-cpp-v2/issues/394#issuecomment-1058431481 from another AWS repo).

Perhaps the documentation of `USE_OPENSSL` being "Set this if you want to use _your system's_ OpenSSL 1.0.2/1.1.1 compatible libcrypto" (emphasis mine) is misleading. In CI [it found the OpenSSL library from the Conda prefix](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=782210&view=logs&j=656edd35-690f-5c53-9ba3-09c10d0bea97&t=986b1512-c876-5f92-0d81-ba851554a0a3&l=185).

**Comment by h-vetinari** [comment]
Hm. Not sure what they mean by "bring your own crypto", but I guess the system openssl does not fall under "our own". In any case, even without `USE_OPENSSL` we're [looking for](https://github.com/awslabs/aws-c-cal/blob/e4c50672c2cc4a6cd8904b79c8c0d7255d2d67f4/cmake/aws-c-cal-config.cmake#L13) `crypto`, and actually aws-c-cal will install its own [`FindCrypto`](https://github.com/awslabs/aws-c-cal/blob/main/cmake/modules/Findcrypto.cmake) file. This depends quite essentially on openssl AFAICT, so even though the link check appears OK without openssl, I'm guessing that actually running a CMake example that does `find_package(aws-c-cal)` might potentially fail.

If you want to go ahead with this PR, please add a cmake_test (see [here](https://github.com/conda-forge/libevent-feedstock/tree/main/recipe/cmake_test) for an example; or a more [elaborate](https://github.com/conda-forge/abseil-cpp-feedstock/blob/main/recipe/cmake_test/CMakeLists.txt) one)

**Comment by teo-tsirpanis** [comment]
I added a CMake test. It fails on Apple Silicon with:

```
/Users/runner/miniforge3/conda-bld/aws-c-cal_1695034368080/test_tmp/run_test.sh: line 10:
/Users/runner/miniforge3/conda-bld/aws-c-cal_1695034368080/_test_env_[…]/bin/cmake: Bad CPU type in executable
```

Looks like it is invoking the x64 CMake. Did I do anything wrong in the test?

**Comment by h-vetinari** [comment]
> I added a CMake test.

Cool, nicely done. However, I'm still skeptical that this works without openssl present, and as it happens, the test doesn't validate that for us, as we're now pulling in openssl as a transitive dependency from somewhere:
```
The following NEW packages will be INSTALLED:

    aws-c-cal:       0.6.2-h17ab6e7_1        local      
    aws-c-common:    0.9.0-hcfcfb64_0        conda-forge
    bzip2:           1.0.8-h8ffe710_4        conda-forge
    ca-certificates: 2023.7.22-h56e8100_0    conda-forge
    cmake:           3.27.5-hf0feee3_0       conda-forge
    gettext:         0.21.1-h5728263_0       conda-forge
    krb5:            1.21.2-heb0366b_0       conda-forge
    libcurl:         8.3.0-hd5e4a3a_0        conda-forge
    libexpat:        2.5.0-h63175ca_1        conda-forge
    libffi:          3.4.2-h8ffe710_5        conda-forge
    libglib:         2.78.0-he8f3873_0       conda-forge
    libiconv:        1.17-h8ffe710_0         conda-forge
    libssh2:         1.11.0-h7dfc565_0       conda-forge
    libuv:           1.44.2-hcfcfb64_1       conda-forge
    libzlib:         1.2.13-hcfcfb64_5       conda-forge
    ninja:           1.11.1-h91493d7_0       conda-forge
    openssl:         3.1.2-hcfcfb64_0        conda-forge  <-----
    [...]
```

On the other hand, having the test is already an improvement, so even if that transitive openssl were to fall away (_and_ it turns out we still need it), then at least the CI should be able to tell us.

> It fails on Apple Silicon with:

That's completely expected, I'll push the required fix.

**Review by h-vetinari — APPROVED** [review]
Thanks for the PR!

File: recipe/cmake_test/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,4 @@
+project(cf_dummy LANGUAGES C CXX)
```
Comment (h-vetinari) [inline recipe/cmake_test/CMakeLists.txt:1]:
```suggestion
project(cf_dummy LANGUAGES C)

```

**Pull request merged**